DfE.FindInformationAcademiesTrusts/Services/Export/ExportService.cs

```diff
@@ -96,46 +101,202 @@ private static byte[] GenerateSpreadsheet(
+
+            SetTextCell(worksheet, i + 4, 17, percentageFull > 0 ? $"{percentageFull}%" : string.Empty);
+            SetTextCell(worksheet, i + 4, 18,
                 freeSchoolMealsData?.PercentageFreeSchoolMeals.HasValue == true
                     ? $"{freeSchoolMealsData.PercentageFreeSchoolMeals}%"
                     : string.Empty
-            };
+            );
+        }
+
+        worksheet.Columns().AdjustToContents();
+        return SaveWorkbookToByteArray(workbook);
+    }
+
+    public async Task<byte[]> ExportOfstedDataToSpreadsheetAsync(string uid)
+    {
+        var headers = new List<string>
+        {
+            "School Name",
+            "Date Joined",
+            "Date of Current Inspection",
+            BeforeOrAfterJoiningHeader,
+            "Quality of Education",
+            "Behaviour and Attitudes",
+            "Personal Development",
+            "Leadership and Management",
+            "Early Years Provision",
+            "Sixth Form Provision",
+            "Date of Previous Inspection",
+            "Previous Quality of Education",
+            "Previous Behaviour and Attitudes",
+            "Previous Personal Development",
+            "Previous Leadership and Management",
+            "Previous Early Years Provision",
+            "Previous Sixth Form Provision",
+            BeforeOrAfterJoiningHeader,
+            "Effective Safeguarding",
+            "Category of Concern"
+        };
+
+        var trustSummary = await trustRepository.GetTrustSummaryAsync(uid);
+        var academiesDetails = await academyRepository.GetAcademiesInTrustDetailsAsync(uid);
+        var academiesOfstedRatings = await academyRepository.GetAcademiesInTrustOfstedAsync(uid);
 
-            for (var j = 0; j < rowData.Length; j++)
-            {
-                worksheet.Cell(i + 4, j + 1).SetValue(rowData[j]);
-            }
+        return GenerateOfstedSpreadsheet(trustSummary, academiesDetails, headers, academiesOfstedRatings);
+    }
+
+    private static byte[] GenerateOfstedSpreadsheet(
+       TrustSummary? trustSummary,
+       AcademyDetails[] academies,
+       List<string> headers,
+       AcademyOfsted[] academiesOfstedRatings)
+    {
+        using var workbook = new XLWorkbook();
+        var worksheet = workbook.Worksheets.Add("Ofsted");
+
+        WriteTrustInformation(worksheet, trustSummary);
+        WriteHeaders(worksheet, headers, 3);
+
+        for (var i = 0; i < academies.Length; i++)
+        {
+            var academyDetails = academies[i];
+            var urn = academyDetails.Urn;
+            var ofstedData = academiesOfstedRatings.SingleOrDefault(x => x.Urn == urn);
+
+            var previousRating = ofstedData?.PreviousOfstedRating ?? OfstedRating.NotInspected;
+            var currentRating = ofstedData?.CurrentOfstedRating ?? OfstedRating.NotInspected;
+
+            // School Name
+            SetTextCell(worksheet, i + 4, 1, academyDetails.EstablishmentName ?? string.Empty);
```

Comment:
The row number should be calculated as a variable at the top of the function and then reused